kartothek/utils/__init__.py

```diff
@@ -0,0 +1,5 @@
+"""
+Utilities that are not klee-specific but are required to archive certain tasks.
```

Comment:
1. drop klee reference


2. What's our intention regarding versioning of the `utils` module? We currently claim to follow semver for `io` and `serialization`, see https://github.com/JDASoftwareGroup/kartothek/blob/master/docs/versioning.rst
We should extend this document with the `utils` as long as we don't have a well maintained `api` module

Reply:
I agree with with the versioning. @fjetter if you have a better proposal for where the `utils` modules should live, please speak up. I couldn't find a nice place. It seems that ktk up until now puts everything either in some `core` submodule or in `io_components`.

Reply:
Added `api` to main package and `utils` to subpackage, as discussed with @marco-neumann-by .